pyuvis/io.py

```diff
@@ -47,28 +45,137 @@ def __init__(self, fname):
+        Returns
+        ----------
+        np.array of np.double values in nanometers (nm)
+            The wavelengths for each spectral bin of the QUBE observation
+        """
+
+        if (self.wavelengths is None):
+            # The Cal matrix LBL file BAND_BIN_CENTER field does not take into account
+            # spectral binning.  Therefore, we must take that into account here.
+            if (self.cal_matrix is not None):
+                wavelengths = np.array(self.cal_matrix.cubelabel['BAND_BIN_CENTER'], dtype=np.double)
+                if (self.BAND_BIN > 1):
+                    binned_wavelengths = np.zeros((self.n_bands,), dtype=wavelengths.dtype)
+                    for iwave in range(0, wavelengths.size, self.BAND_BIN):
+                        binned_wavelengths[iwave // self.BAND_BIN] = np.mean(wavelengths[iwave:iwave+self.BAND_BIN])
+                    wavelengths = binned_wavelengths / 10.0 # Convert from Angstroms to nm
+            else:
+                # If calibration matrix is not present, set to default wavelengths.
+                wavelengths = np.linspace(self.wave_min, self.wave_max, self.shape[0])
+
+            self.wavelengths = wavelengths * u.nm
+
+        return self.wavelengths
 
     @property
     def xarray(self):
         return xr.DataArray(self.data)
 
+class UV_PDS(QUBE):
+    """
+    Class: UV_PDS inherits from QUBE and is intended as a base class for
+        FUV_PDS and EUV_PDS
+    """
 
-class FUV_PDS(QUBE):
-    wave_min = 111.5 * u.nm
-    wave_max = 190.0 * u.nm
+    def __init__(self, fname):
+        """
+        Constructor for UV_PDS
+
+        Parameters
+        ----------
+        fname : str
+            Filename for the PDS QUBE to be read.
+        """
+
+        # Call superclass constructor
+        super().__init__(fname)
+
+        # Try to load the calibration Matrix file.  If not present, set to None.
+        self.cal_label_fname = self.path.joinpath(self.path.parent, self.file_id + '_CAL_3.LBL')
+        self.cal_data_fname = self.path.joinpath(self.path.parent, self.file_id + '_CAL_3.DAT')
+        if (self.cal_label_fname.exists() and self.cal_data_fname.exists()):
+            self.cal_matrix = QUBE(self.cal_data_fname)
+            waves = self.waves
+            self.wave_min = np.min(waves)
+            self.wave_max = np.max(waves)
+        else:
+            self.cal_matrix = None
+
+class FUV_PDS(UV_PDS):
+    """
+    Class: FUV_PDS defines an FUV QUBE object.
+    """
+
+    def __init__(self, fname):
+        """
+        Constructor for FUV_PDS
+
+        Parameters
+        ----------
+        fname : str
+            Filename for the PDS QUBE to be read.
+        """
 
+        # Call superclass constructor
+        super().__init__(fname)
+
+        # If calibration matrix file was not present (loaded by the superclass)
+        # then set to default wavelength range for FUV.
+        if (self.cal_matrix is None):
+            self.cal_matrix = None
```

Comment:
not seeing how this line is required as it's only reached when cal_matrix is None in the first place?

Reply:
Ah, yes.  I see your point.  I'll remove that line.